go-controller/pkg/node/routemanager/route_manager.go

```diff
@@ -353,5 +343,21 @@ func RoutePartiallyEqual(r, x netlink.Route) bool {
 		r.Gw.Equal(x.Gw) &&
 		r.Table == x.Table &&
 		r.Flags == x.Flags &&
-		r.MTU == x.MTU
+		r.MTU == x.MTU &&
+		r.Type == x.Type &&
+		r.Priority == x.Priority &&
+		r.Scope == x.Scope
+}
```

Comment:
Actually, I am a bit unsure why are we looking at all these fields. `netlinkDelRoute` only looks at `LinkIndex`, `Dst` and `Table` and we shnould probably stick to that here to make our lives way easier.
Additionally, we should use and filter by `Protocol` static to avoid handling routes added by other protocols (i.e. BGP)
Let's bring up this discussion with @martinkennelly

Reply:
because those fields are used by consumers of route manager and it was a lazy way check.

Reply:
That doesn't really address the concern I have here. Let me explain better.

Look at `delRoute` method:
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/27033e321c7e881a2a5a356b062ceee796b7d50d/go-controller/pkg/node/routemanager/route_manager.go#L117-L150

It fist calls `netlinkDelRoute` which will delete routes based on `LinkIndex`, `Dst` and `Table` only. It then calls `RoutePartiallyEqual` to remove from cache which considers all these fields.

So it might happen that we delete routes with netlink that we then don't delete from the cache. What sense does that make?

Reply:
So the question is: should we alight these two to the minimum common denominator of fields?